src/main/java/wolox/reactortraining/services/BotService.java

```diff
@@ -38,4 +66,76 @@
             .bodyToMono(BotResponse.class);
     }
 
+    @Override
+    public Mono<Void> createBot(BotCreationDto botCreationDto) {
+        Mono<User> userMono = createUser(botCreationDto.getUsername());
+        Mono<Void> createBot = createBot(botCreationDto.getBotName(), botCreationDto.getTopics());
+
+        return userMono
+            .flatMap(user ->
+                Mono.zip(createTopics(user, botCreationDto.getTopics()), Mono.just(user)))
+            .flatMap(tuple -> {
+                User user = tuple.getT2();
+                List<Topic> topics = tuple.getT1();
+
+                user.addTopics(topics);
+
+                return userRepository.save(user);
+            })
+            .then(createBot);
+    }
+
+    private Mono<User> createUser(String username) {
+        UserDto userDto = new UserDto();
+        userDto.setUsername(username);
+
+        User user = new User(userDto);
+
+        return userRepository.insert(user);
+    }
+
+    private Mono<List<Topic>> createTopics(User user, List<String> topics) {
+        return Flux
+            .fromIterable(topics)
+            .map(description -> {
+                TopicDto topicDto = new TopicDto();
```

Comment:
this can be simplified: TopicDto::new

Reply:
It can't be because we are invoking another object's method